projects/schematics/src/migrations/mechanism/rename-symbol/rename-symbol.ts

```diff
@@ -0,0 +1,77 @@
+import { Tree } from '@angular-devkit/schematics';
+import { ImportDeclarationStructure } from 'ts-morph';
+import { RenamedSymbol } from '../../../shared/utils/file-utils';
+import { createProgram } from '../../../shared/utils/program';
+import { getProjectTsConfigPaths } from '../../../shared/utils/project-tsconfig-paths';
+import { getDefaultProjectNameFromWorkspace } from '../../../shared/utils/workspace-utils';
+
+export function migrateRenamedSymbols(
+  tree: Tree,
+  renamedSymbols: RenamedSymbol[]
+): Tree {
+  const project = getDefaultProjectNameFromWorkspace(tree);
+
+  const { buildPaths } = getProjectTsConfigPaths(tree, project);
+  const basePath = process.cwd();
+
+  for (const tsconfigPath of buildPaths) {
+    const { appSourceFiles } = createProgram(tree, basePath, tsconfigPath);
+    appSourceFiles.map((sourceFile) => {
+      const importDeclarationStructures: ImportDeclarationStructure[] = [];
+
+      sourceFile.getImportDeclarations().forEach((id) => {
+        id.getImportClause()
+          ?.getNamedImports()
+          .forEach((namedImport) => {
+            const importName = namedImport.getName();
+
+            const renamedSymbol = renamedSymbols.find(
+              (_) =>
+                _.previousNode === importName &&
+                _.previousImportPath ===
+                  id
+                    .getModuleSpecifier()
+                    .getText()
+                    .substr(1, id.getModuleSpecifier().getText().length - 2)
+            );
+
+            if (renamedSymbol) {
+              const newNodeName = renamedSymbol.newNode
```

Comment:
Maybe I don't fully understand the logic, but in case the `newNode` config is not provided, shouldn't we fallback to the `previousNode`?